### ISSUE TITLE
Correctly validate new CSS 4 hex notations

### ISSUE THREAD

**Issue by shroudedcode** [sasstools/sass-lint#1179]
The new  `#RRGGBBAA` and `#RGBA` notations for hexadecimal colors with alpha channel from [CSS Color Module Level 4](https://drafts.csswg.org/css-color/#hex-notation) are now [supported by most browsers](https://caniuse.com/#feat=css-rrggbbaa) and popular tools (like VSCode).

`sass-lint` still displays an error:

![sass-lint showing the error](https://user-images.githubusercontent.com/29176678/35985548-327c7a16-0cf7-11e8-9190-2b9698bbca05.png)

The `no-invalid-hex` rule should be updated to allow the new notations.

---

To implement this, the [`isValidHex` function](https://github.com/sasstools/sass-lint/blob/be43c0bd231774873668b0560a8a893d89ef10d0/lib/helpers.js#L184-L189) would have to be slightly changed:

```js
helpers.isValidHex = function (str) {
  return /^[a-f0-9]{8}|[a-f0-9]{6}|[a-f0-9]{4}|[a-f0-9]{3}$/i.test(str);
};
```

*I'll open a pull request if you approve of the change.*
